### DISCUSSION TITLE
My vcpkg repo is getting oversized.

### DISCUSSION THREAD

**Discussion by KRJackLee** [microsoft/vcpkg#28889] [Q&A]
My vcpkg repo was just 1.6GB with a few packages. Then I installed `boost:x64-windows`, it became 16GB. After installing `quantlib:x64-windows-static`, `/vcpkg` becomes 26GB. I knew that installing `boost` and `quantlib` involves many other dependencies, but under `/vcpkg/installed` there isn't much space taken. So I wonder whether there are caches that can be deleted after installation success? For example, `/vcpkg/buildtrees` takes like half of the space of `/vcpkg`. and `/vcpkg/packages` takes a few GBs also. Can I delete the files within?

**Comment by StarGate-One** [comment]
One can delete ./buildtrees ./downloads ./packages directories/folders depending on your needs. 
In the future when running vcpkg install, you can add one or more of the following to the command line:
`--clean-after-build`                       Clean buildtrees, packages and downloads after building each package
`--clean-buildtrees-after-build`     Clean buildtrees after building each package
`--clean-packages-after-build`      Clean packages after building each package
`--clean-downloads-after-build`   Clean downloads after building each package

I personally use the 2 options to delete the buildtrees and packages, as I want to keep the downloads so I do not need to download the sources again and periodically cleanup the downloads as new versions of sources are downloaded.

Do not use the `--clean-packages-after-build` or `--clean-after-build` if you plan to use the `vcpkg export` command as vcpkg needs these to build the exported packages.

The `--clean-buildtrees-after-build` will leave the logs behind, but the original uncompressed source tree, triplet-debug and triplet-release will be deleted upon successful build, compile, link and install of each port is completed.

If `--clean-after-build`  is used, each downloaded source, in addition to the above stated buildtrees and packages are deleted upon successful install of the port.

**Reply by autoantwort** [reply]
Imho these command line parameters could also be the default.

**Comment by JackBoosY** [comment]
You can simply delete _buildtrees_ and _packages_ folder.
